pom.xml

```diff
@@ -35,6 +43,17 @@
       </properties>
     <build>
     <plugins>
+          <plugin>
+            <groupId>com.github.inodb.maven-external-version</groupId>
+            <artifactId>maven-external-version-plugin</artifactId>
+            <version>b7bc6a6ff7d51295089d5dc7ff64084063440cc6</version>
+            <extensions>true</extensions>
+            <configuration>
+              <strategy hint="script">
+                <script>./version.sh</script>
```

Comment:
Do you know at which phase of the Maven build lifecycle the script will run? I've been trying to use `mvn dependency:go-offline` to download at least some of the dependencies and plugins separately before building the actual source code, and if it makes sense, I believe it would be convenient for my automation if `version.sh` wouldn't be invoked until a later step.

Reply:
Annoyingly, it looks like this script *does* run earlier in the build lifecycle, which would leave one of the workflows I maintain in `thehyve/cbioportal-docker` (making a PR to `cBioPortal/cbioportal` is backlogged for a less-busy time of year) unable to use the new feature.

Reply:
@fedde-s hmm that is tricky - im not sure if it's possible to change it to a different lifecycle. One thing we perhaps could do is to allow disabling the plugin completely. Then instead of using this script, you could do a find & replace on `0-unknown-version-SNAPSHOT`. Since we won't change the version in the pom anymore, you could say this is a new feature as well ;)?

Reply:
I mostly went for a maven approach in this PR, because I figured this way developers/deployers don't have to change their existing workflow. It is much easier imo to do a find and replace before running maven, but figured it be nicer not to request people to type a different cmd when compiling

Reply:
For now, I can just make that one Dockerfile use-case hard-code the `PROJECT_VERSION` to a fixed string. It would be very useful to be able to see the commit hash here, but only if it can be read *after* dependencies are cached. I think that feature should go on a backlog and not block your PR though, as only a few people actually use that Dockerfile as far as I'm aware.